src/services/codefixes/convertPromisesToAwaitAndAsync.ts

```diff
@@ -0,0 +1,161 @@
+namespace ts.codefix {
+    const fixId = "convertPromisesToAwaitAndAsync";
+    const errorCodes = [Diagnostics.This_may_be_converted_to_use_async_and_await.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context: CodeFixContext) {
+            const changes = textChanges.ChangeTracker.with(context, (t) => convertToAsyncAwait(t, context.sourceFile, context.span.start, context.program.getTypeChecker()));
+            return [createCodeFixAction(fixId, changes, Diagnostics.Convert_to_use_async_and_await, fixId, Diagnostics.Convert_all_to_use_async_and_await)];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, err) => convertToAsyncAwait(changes, err.file!, err.start, context.program.getTypeChecker())),
+    });
+    function convertToAsyncAwait(changes: textChanges.ChangeTracker, sourceFile: SourceFile, position: number, checker: TypeChecker): void {
+        // get the function declaration - returns a promise
+        const funcToConvert: FunctionLikeDeclaration = checker.getSymbolAtLocation(getTokenAtPosition(sourceFile, position, /*includeEndPosition*/ false)).valueDeclaration as FunctionLikeDeclaration;
+        // add the async keyword
+        changes.insertModifierBefore(sourceFile, SyntaxKind.AsyncKeyword, funcToConvert);
+
+        const stmts: NodeArray<Statement> = (<FunctionBody>funcToConvert.body).statements;

```

Comment:
call the same function you had in suggestionDiagnostics.ts to ensure we are doing the same thing in both places. if the function returns you an empty set, return undefined immediately.